### ISSUE TITLE
Deleting a node (S * *) after dropping a predicate will re-create the dropped predicate with default type

### ISSUE THREAD

**Issue by sgon00** [dgraph-io/dgraph#1965]
Hi, I tried the go client example and it has the `married` schema defined:

```
married: bool .
```

when I do `schema{}` query, I found out:

It is `"predicate": "married", "type": "default"` in dgraph 1.0.1 and master 52994ae301774538642295e4e65096ba38f024de.

But it is `"predicate": "married", "type": "bool"` in dgraph 0.9.4

Is this a bug or expected in latest dgraph?

**Comment by pawanrawal** [comment]
Hey @sgon00 

I was not able to reproduce this. Could you share the Go client example which can be used to reproduce this? If the alter operation was successful then schema should have been changed to `bool`. Dgraph server should also print a log about the schema change if it goes through.

**Comment by sgon00** [comment]
@pawanrawal 

Sorry about the wrong issue description. I did some more tests on this issue and changed this issue's title to `Deleting a node (S * *) after dropping a predicate will re-create the dropped predicate with default type`
I am not sure if this is considered as a bug or not. If it's expected, you can simply close it. The following is the complete test code:

```go
package main

import (
	"context"
	"encoding/json"
	"fmt"
	"log"

	"github.com/dgraph-io/dgraph/client"
	"github.com/dgraph-io/dgraph/protos/api"
	"google.golang.org/grpc"
)

type person struct {
	UID     string `json:"uid,omitempty"`
	Name    string `json:"name,omitempty"`
	Age     int    `json:"age,omitempty"`
	Married bool   `json:"married,omitempty"`
}

func main() {
	conn, err := grpc.Dial("127.0.0.1:9080", grpc.WithInsecure())
	if err != nil {
		log.Fatal("While trying to dial gRPC")
	}
	defer conn.Close()

	dg := client.NewDgraphClient(api.NewDgraphClient(conn))

	err = dg.Alter(context.Background(), &api.Operation{DropAll: true})
	handleErr(err)

	// set schema
	err = dg.Alter(context.Background(), &api.Operation{
		Schema: `
		name: string @index(exact) .
		age: int .
		married: bool .
		`,
	})
	handleErr(err)

	p1 := person{Name: "Alice", Age: 26, Married: true}

	pb, err := json.Marshal(p1)
	handleErr(err)

	// insert data
	assigned, err := dg.NewTxn().Mutate(context.Background(), &api.Mutation{CommitNow: true, SetJson: pb})
	handleErr(err)
	aliceUID := assigned.Uids["blank-0"]
	//fmt.Println(aliceUID)

	qSchema := `schema {
		predicate
		type
	}`
	resp, err := dg.NewTxn().Query(context.Background(), qSchema)
	handleErr(err)
	fmt.Println("\nQuery schema: ", resp.String(), "\n")

	// Drop married predicate
	err = dg.Alter(context.Background(), &api.Operation{DropAttr: "married"})
	handleErr(err)

	resp, err = dg.NewTxn().Query(context.Background(), qSchema)
	handleErr(err)
	fmt.Println("\nQuery schema after dropping married predicate: ", resp.String(), "\n")

	alice := person{
		UID: aliceUID,
	}
	aliceb, err := json.Marshal(alice)
	handleErr(err)

	// Delete alice, perform (S * *) Deletion
	_, err = dg.NewTxn().Mutate(context.Background(), &api.Mutation{CommitNow: true, DeleteJson: aliceb})
	handleErr(err)

	resp, err = dg.NewTxn().Query(context.Background(), qSchema)
	handleErr(err)
	fmt.Println("\nQuery schema after deleting a node (S * * deletion): ", resp.String(), "\n")

}

func handleErr(err error) {
	if err != nil {
		log.Fatal(err)
	}
}
```

the output:

```
Query schema:  json:"{}" schema:<predicate:"_predicate_" type:"string" > schema:<predicate:"name" type:"string" > schema:<predicate:"age" type:"int" > schema:<predicate:"married" type:"bool" > txn:<start_ts:151 lin_read:<> > latency:<parsing_ns:543 processing_ns:1082 encoding_ns:511792 >  

Query schema after dropping married predicate:  json:"{}" schema:<predicate:"_predicate_" type:"string" > schema:<predicate:"name" type:"string" > schema:<predicate:"age" type:"int" > txn:<start_ts:152 lin_read:<> > latency:<parsing_ns:503 processing_ns:1031 encoding_ns:551492 >  

Query schema after deleting a node (S * * deletion):  json:"{}" schema:<predicate:"_predicate_" type:"string" > schema:<predicate:"age" type:"int" > schema:<predicate:"married" type:"default" > schema:<predicate:"name" type:"string" > txn:<start_ts:155 lin_read:<> > latency:<parsing_ns:428 processing_ns:895 encoding_ns:399635 > 
```

**Comment by pawanrawal** [comment]
Hey @sgon00 

Thanks for the reproducible example. What is happening here is that schema for `married` is being automatically derived again while doing `S * *` deletion, hence it gets a `default` type. We shouldn't be deriving the schema for delete mutations. I will work on a fix.

**Comment by pawanrawal** [comment]
A fix was merged to master with f397e76f186c70d35974465e8615e381f047fd74 and would be part of the nightly and next release.
